test/inductor/test_compiled_optimizers.py

```diff
@@ -208,6 +200,9 @@ def test_fn(self):
             if run_cudagraphs:
                 stack.enter_context(config.patch({"triton.cudagraphs": True}))
 
+            if isinstance(kwargs.get("lr", None), torch.Tensor):
```

Comment:
Does this change suggest that the responsibility would be on the user to move their lr to the right device? What happens if there are mixed devices among the parameters of the optimizer?

Reply:
Basically it just can't be CPU (which I'm not sure we test in the optimizer cudagraphs tests). We could possibly add something to the compiler cudagraphs impl to detect that they're CPU and 0-D but currently we basically print a warning that cudagraphs didn't run, and the line that causes it.